app/questionnaire/questionnaire_schema.py

```diff
@@ -47,7 +47,10 @@ def get_answer_dependencies_by_id(self, answer_id):
         return self._answer_dependencies.get(answer_id, [])
 
     def get_group_and_block_id_by_answer_id(self, answer_id):
-        return self._answer_with_block_and_group_ids.get(answer_id)
+        return self._answer_with_block_and_group_id.get(answer_id)
+
+    def get_section_and_group_id_by_block_id(self, block_id):
```

Comment:
The `piped-answers-in-repeating-groups-fix` branch patches a `parent_id` onto blocks and groups. Once that's merged that may by a better solution for finding the parent section.

I.e. (in `PathFinder.get_next_location`)
```
location_group = self.schema.get_group(current_block_id['parent_id'])
location_section = self.schema.get_section(location_group['parent_id'])
```

If we merge this first then consider altering this in the other branch. @dwyeradam

Reply:
I'll have a look at that then.